File: IWB2B/Code/Java/Dev/B2BIntegrationService/src/main/java/com/iweb2b/api/integration/model/auth/AuthTokenRequest.java
package com.iweb2b.api.integration.model.auth;

import javax.validation.constraints.NotEmpty;

import lombok.Data;

@Data
public class AuthTokenRequest {

	@NotEmpty(message = "API Name cannot be null or empty.")
	private String apiName;
	
	@NotEmpty(message = "Client ID cannot be null or empty.")
	private String clientId;
	
	@NotEmpty(message = "Client Secret Key cannot be null or empty.")
	private String clientSecretKey;
	
	@NotEmpty(message = "GrantType cannot be null or empty.")
	private String grantType;
	
	@NotEmpty(message = "OAuth Username cannot be null or empty.")
	private String oauthUserName;
	
	@NotEmpty(message = "OAuth Password cannot be null or empty.")
	private String oauthPassword;
}

File: MobileApp/Clara Timekeeping Android/app/src/main/java/com/clara/timekeeping/model/request/AuthTokenRequest.kt
package com.clara.timekeeping.model.request

import com.google.gson.annotations.SerializedName

data class AuthTokenRequest(
    @SerializedName("clientId") val clientId: String? = null,
    @SerializedName("clientSecretKey") val clientSecretKey: String? = null,
    @SerializedName("grantType") val grantType: String? = null,
    @SerializedName("oauthPassword") val oauthPassword: String? = null,
    @SerializedName("oauthUserName") val oauthUserName: String? = null,
    @SerializedName("apiName") val apiName: String? = null
)
